holmes/plugins/toolsets/infrainsights/infrainsights_client.py

```
@@ -0,0 +1,205 @@
+    
+    def _make_request(self, method: str, endpoint: str, **kwargs) -> Dict[str, Any]:
+        """Make HTTP request to InfraInsights API"""
+        url = f"{self.config.base_url.rstrip('/')}/{endpoint.lstrip('/')}"
+        
+        try:
+            response = self.session.request(method, url, **kwargs)
+            response.raise_for_status()
+            return response.json()
+        except requests.exceptions.RequestException as e:
+            logging.error(f"InfraInsights API request failed: {e}")
+            raise Exception(f"Failed to connect to InfraInsights API: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)